doc/reference/configuration/configuration_reference.rst

```diff
@@ -3599,6 +3599,290 @@ The ``replication`` section defines configuration parameters related to :ref:`re
+            enabled: true
+            by: prefix
+            prefix: '{{ replicaset_name }}'
+
+    |
+    | Type: string
+    | Default: ``nil``
+    | Environment variable: TT_REPLICATION_AUTOEXPEL_BY
+
+
+
+replication.autoexpel_enabled.*
+~~~~~~~~~~~~~
+
+    The ``replication.autoexpel_enabled`` field is a boolean configuration option that determines whether the autoexpel logic is active for the cluster.
+    This feature is designed to automatically manage dynamic cluster configurations by removing instances that are no longer present in the YAML configuration.
+
+    ..  NOTE::
+
+        By default, the ``enabled`` field is set to ``false``, meaning the ``autoexpel`` logic is turned off. This ensures that no instances are automatically removed unless explicitly configured.
+
+    Enabling ``autoexpel`` logic
+
+    To enable ``autoexpel``, you should set enabled to true in the ``replication.autoexpel`` section of your YAML configuration:
+
+    .. code-block:: yaml
+
+        replication:
+          autoexpel:
+            enabled: true
+            by: prefix
+            prefix: '{{ replicaset_name }}'
+
+
+    To disable ``autoexpel``, set enabled to ``false``.
+
+
+    Dependencies
+
+    If ``enabled`` is set to ``true``, the following fields are required:
+
+        1.	``by``: specifies the criterion for ``autoexpel`` (e.g., ``prefix``).
+        2.	``prefix``: defines the pattern used to match instance names for expulsion.
+
+    Failure to configure these fields when enabled is true will result in a configuration error.
+
+    |
+    | Type: boolean
+    | Default: ``false``
+    | Environment variable: TT_REPLICATION_AUTOEXPEL_ENABLED
+
+
+replication.autoexpel_prefix.*
+~~~~~~~~~~~~~
+
+    The ``prefix`` field filters instances for expulsion by differentiating cluster instances (from the YAML configuration) from external services (e.g., CDC tools). Only instances matching the prefix are considered.
+
+    A consistent naming pattern ensures the ``_cluster`` system space automatically aligns with the YAML configuration.
+
+
+    If the ``prefix`` field is not set (``nil``), the ``autoexpel`` logic cannot identify instances for expulsion, and the feature will not function.
+    This field is **mandatory** when ``replication.autoexpel_enabled`` is set to ``true``.
+
+
+    How it works:
+
+	1.	The prefix filters instance names (e.g., ``{{ replicaset_name }}`` for replicaset-specific names or ``i-`` for names starting with ``i-``).
+	2.	Instances matching the prefix and removed from the YAML configuration are expelled.
+	3.	Unnamed instances or those not matching the prefix are ignored.
+
+    Dynamic prefix based on replicaset name:
+
+    .. code-block:: yaml
+
+        replication:
+          autoexpel:
+            enabled: true
+            by: prefix
+            prefix: '{{ replicaset_name }}'
+
+
+    In this setup:
+
+	- Instances are grouped by replicaset names (e.g., ``r-001-i-001`` for ``replicaset r-001``).
```

Comment:
We have `groups` configuration level, so I'm in some doubt regarding using the 'grouped' word. Possibly may be confusing. Can we just say 'prefixed with a replicaset name', 'named according to a replicaset name' or something of this kind?

Reply:
fixed